myapp/logic/trade_logic.py

```diff
@@ -0,0 +1,34 @@
+from myapp import stock_api
+from myapp import models
+
+def extract_info(params):
+    try:
+        number_of_stocks = int(params["number_of_stocks"])
+        if number_of_stocks <= 0:
+            raise Exception
+        stock_info = stock_api.get_stock_info(params["stock_selector"])
```

Comment:
better to unpack whatever you need at the start of the function. 
you will defintley run into unexpected, hard to catch KeyErrors like this.

Reply:
Now the function take each input separately and not as a dictionary, problem solved